find_packages with exclude here doesn't completely exclude the test package.
Following the documentation using ['*.test.*','test.*','test'] would prevent 
the test package and its sub-packages from being installed.

For reference, follow this url:
http://setuptools.readthedocs.io/en/latest/setuptools.html#using-find-packages

diff -ru jedi-0.11.1/setup.py jedi-0.11.1_modified/setup.py
--- jedi-0.11.1/setup.py	2017-12-14 21:51:15.000000000 +0000
+++ jedi-0.11.1_modified/setup.py	2017-12-22 19:42:31.798180651 +0000
@@ -32,7 +32,7 @@
       license='MIT',
       keywords='python completion refactoring vim',
       long_description=readme,
-      packages=find_packages(exclude=['test']),
+      packages=find_packages(exclude=['*.test.*','test.*','test']),
       install_requires=install_requires,
       extras_require={'dev': ['docopt']},
       package_data={'jedi': ['evaluate/compiled/fake/*.pym']},
